### PR TITLE
:bug: Fix "fast check in" button not correctly selecting last stop if train end prematurely

### PR DESCRIPTION
**Pull request by DRSchlaubi**
Seemingly, db rest also cancels the last stop of a train; therefore we also check for departure/arrival for the trip station list but we don't for the fast check in

https://github.com/Traewelling/traewelling/blob/f3ab90f01ff09281293eb723302baaa0f7f6c833/resources/views/admin/checkin/trip.blade.php#L30

### CODE REVIEW

**Review by jeyemwey — APPROVED** [review]
Looks good to me. Thanks!

**Comment by jeyemwey** [comment]
@MrKrisKrisu Somehow, the tests don't run through and thus I cannot merge it. Can you try that please?

If at all possible, I'd like to rebase the changes onto the current head to disguise the second metro line that comes from the earlier parent.

File: app/Http/Controllers/FrontendTransportController.php [hunk 1]
```diff
@@ -121,7 +121,7 @@ public function TrainTrip(Request $request): Renderable|RedirectResponse {
 
         // Find out where this train terminates and offer this as a "fast check-in" option.
         $terminalStopIndex = count($TrainTripResponse['stopovers']) - 1;
-        while ($terminalStopIndex >= 1 && @$TrainTripResponse['stopovers'][$terminalStopIndex]['cancelled'] == true) {
+        while ($terminalStopIndex >= 1 && @$this->is_cancelled($TrainTripResponse['stopovers'][$terminalStopIndex])) {
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/FrontendTransportController.php]:
```suggestion
        while ($terminalStopIndex >= 1 && @$this->isCancelled($TrainTripResponse['stopovers'][$terminalStopIndex])) {
```

File: app/Http/Controllers/FrontendTransportController.php [hunk 2]
```diff
@@ -225,4 +225,8 @@ public function setTrainHome(Request $request): RedirectResponse {
             return redirect()->back()->with(['error' => __('messages.exception.generalHafas')]);
         }
     }
+
+    private function is_cancelled(mixed $param): bool {
```
Comment (MrKrisKrisu) [inline app/Http/Controllers/FrontendTransportController.php]:
```suggestion
    private function isCancelled(mixed $param): bool {
```

**Review by MrKrisKrisu — CHANGES_REQUESTED** [review]

**Comment by MrKrisKrisu** [comment]
> @MrKrisKrisu Somehow, the tests don't run through and thus I cannot merge it. Can you try that please?
> 
> If at all possible, I'd like to rebase the changes onto the current head to disguise the second metro line that comes from the earlier parent.

Normally we have a button to accept the action for "first time contributors".... But the button is missing. We can merge this anyway, because we don't have a test for this case.

**Pull request merged**